### ISSUE TITLE
Flag `d` should not be considered valid in ES2021

### ISSUE THREAD

**Issue by fasttime** [eslint-community/regexpp#208]
Regexp flag `d` was introduced in [ES2022](https://github.com/tc39/ecma262/commit/0209d8571b41e1c7dd18befd2ba45312680d6fcd). It should not be considered valid with `ecmaVersion: 2021`, i.e.:

```js
import { RegExpValidator } from "@eslint-community/regexpp";
const validator = new RegExpValidator({ ecmaVersion: 2021 });
validator.validateFlags("d");
```

The above code should throw an error. It did so with version 4.11.2, but in version 4.12.0, no error is thrown.
